cmd/armada/export/logs/exportlogs.go

```diff
@@ -0,0 +1,57 @@
+package logs
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/armada/pkg/defaults"
+	kind "sigs.k8s.io/kind/pkg/cluster"
+)
+
+// ExportLogsFlagpole is a list of cli flags for export logs command
+type ExportLogsFlagpole struct {
+	Clusters []string
+}
+
+// ExportLogsCommand returns a new cobra.Command under export command for armada
+func ExportLogsCommand(provider *kind.Provider) *cobra.Command {
+	flags := &ExportLogsFlagpole{}
+	cmd := &cobra.Command{
+		Args:  cobra.NoArgs,
+		Use:   "logs",
+		Short: "Export kind cluster logs",
+		Long:  "Export kind cluster logs",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// remove existing before exporting
+			_ = os.RemoveAll(filepath.Join(defaults.KindLogsDir, defaults.KindLogsDir))
+
+			var targetClusters []string
+			if len(flags.Clusters) > 0 {
+				targetClusters = append(targetClusters, flags.Clusters...)
+			} else {
+				configFiles, err := ioutil.ReadDir(defaults.KindConfigDir)
+				if err != nil {
+					log.Fatal(err)
+				}
+				for _, configFile := range configFiles {
+					clName := strings.FieldsFunc(configFile.Name(), func(r rune) bool { return strings.ContainsRune(" -.", r) })[2]
+					targetClusters = append(targetClusters, clName)
+				}
```

Comment:
As expected this logic is duplicated across many places in the code. This is the last comment I put about it but you need to move this into a function.